pkg/cloudprovider/providers/gce/cloud/op.go

```diff
@@ -71,7 +79,15 @@ func (o *gaOperation) isDone(ctx context.Context) (bool, error) {
 	if err != nil {
 		return false, err
 	}
-	return op != nil && op.Status == "DONE", nil
+	if op == nil || op.Status != operationStatusDone {
+		return false, nil
+	}
+
+	if op.Error != nil && len(op.Error.Errors) > 0 && op.Error.Errors[0] != nil {
+		e := op.Error.Errors[0]
+		o.err = &GCEOperationError{HTTPStatusCode: op.HTTPStatusCode, Code: e.Code, Message: e.Message}
+	}
+	return true, nil
```

Comment:
Lots of these code are very similar. actually you can embed a struct with an `error` method. When I wrote this, I realized that it also applies to `String()` and `isDone()`. feel free to ignore, totally optional. maybe i will submit a PR later.